hoot-core/src/main/cpp/hoot/core/algorithms/splitter/CornerSplitter.cpp

```diff
@@ -223,15 +228,41 @@ void CornerSplitter::_splitRoundedCorners()
   }
 }
 
-void CornerSplitter::_splitWay(long wayId, long nodeIdx, long nodeId)
+bool CornerSplitter::_splitWay(long wayId, long nodeIdx, long nodeId, bool sharpCorner)
 {
   WayPtr pWay = _map->getWay(wayId);
   if (!pWay)
   {
     LOG_TRACE("way at " << wayId << " does not exist.");
-    return;
+    return false;
+  }
+  //  For sharp corners, some small
+  if (sharpCorner)
+  {
+    GeometryFactory::unique_ptr factory = GeometryFactory::create();
+    //  Check the previous segment to ensure it is larger than the circular error before splitting
+    if (nodeIdx == 1)
+    {
+      boost::shared_ptr<LineString> ls = ElementConverter(_map).convertToLineString(pWay);
+      CoordinateArraySequence* subline = new CoordinateArraySequence();
+      subline->add(0, ls->getCoordinateN(nodeIdx), false);
+      subline->add(1, ls->getCoordinateN(nodeIdx - 1), false);
+      boost::shared_ptr<LineString> sub(factory->createLineString(subline));
+      if (sub->getLength() <= pWay->getCircularError())
+        return false;
+    }
+    //  Check the next segment to ensure it is larger than the circular error before splitting
+    if (nodeIdx == (long)(pWay->getNodeCount() - 2))
+    {
+      boost::shared_ptr<LineString> ls = ElementConverter(_map).convertToLineString(pWay);
+      CoordinateArraySequence* subline = new CoordinateArraySequence();
+      subline->add(0, ls->getCoordinateN(nodeIdx), false);
+      subline->add(1, ls->getCoordinateN(nodeIdx + 1), false);
+      boost::shared_ptr<LineString> sub(factory->createLineString(subline));
+      if (sub->getLength() <= pWay->getCircularError())
+        return false;
+    }
```

Comment:
This is the big change that affects the dog-legs.  It checks if the length of the first or last segment of the split is a dog-leg.  If it is then the way isn't split.  That way it is never left by itself to not conflate when it should.